GCRCatalogs/buzzard.py

```diff
@@ -175,42 +172,25 @@ def _subclass_init(self,
                 'velocity_z': 'truth/VZ',
             }
 
-            if (self.version.split('_')[0] == '1.9.2') | (self.version.split('_')[0] == '2.0.0'):
+            for i, b in enumerate('grizY'):
+                self._quantity_modifiers['mag_{}_des'.format(b)] = (_mask_func, 'truth/OMAG/{}'.format(i))
+                self._quantity_modifiers['magerr_{}_des'.format(b)] = (_mask_func, 'truth/OMAGERR/{}'.format(i))
+                self._quantity_modifiers['mag_true_{}_des'.format(b)] = (_mask_func, 'truth/TMAG/{}'.format(i))
+                self._quantity_modifiers['Mag_true_{}_des_z01'.format(b)] = (_abs_mask_func, 'truth/AMAG/{}'.format(i))
 
-                for i, b in enumerate(['Z_vista', 'Y_vista', 'J_vista', 'H_vista', 'Ks_vista', 'u_lsst', 
+            if not self.version.startswith('1.6'):  # for v1.9.2 and above
+                for i, b in enumerate(['Z_vista', 'Y_vista', 'J_vista', 'H_vista', 'Ks_vista', 'u_lsst',
                                        'g_lsst', 'r_lsst', 'i_lsst', 'z_lsst', 'y_lsst', 'Y_wfirst',
                                        'J_wfirst', 'H_wfirst', 'K_wfirst']):
-
                     i += 6
-
                     self._quantity_modifiers['Mag_true_{}_z0'.format(b)] = (_abs_mask_func, 'auxmag/AMAG/{}'.format(i))
                     self._quantity_modifiers['mag_true_{}'.format(b)] = (_mask_func, 'auxmag/TMAG/{}'.format(i))
                     self._quantity_modifiers['mag_{}'.format(b)] = (_mask_func, 'auxmag/LMAG/{}'.format(i))
 
-            else:
```

Comment:
this looks like a severe delete, but I trust you have removed them for good reasons :-)

Reply:
Yes, thanks for pointing this out -- I should have explained it better.

So for lines where you see `truth/OMAG`, `truth/OMAGERR`, `truth/AMAG`, and `truth/TMAG` -- these for all for DES filters, and they exist for all versions. Hence they are moved to the new lines 175-179. 

For the two lines where you see `lsst/AMAG` and `lsst/TMAG` --- those are only available in v1.9.2 and above, and they have been dealt with in lines 181--188. 

There's one line where it maps `mag_true_x` to `mag_true_x_lsst`. This should be done for all versions, hence I move it to line 190-193. 

I think this explanation also works as me double checking my work --- so thanks for prompting it.

Reply:
Ok thanks Yao!